src/server/StudioManagementSystem.Infrastructure/Repositories/GroupRepositoryAsync.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.EntityFrameworkCore;
+using StudioManagementSystem.Core.Entities;
+using StudioManagementSystem.Infrastructure.DataServices;
+using StudioManagementSystem.Infrastructure.Interfaces.DataServices;
+
+namespace StudioManagementSystem.Infrastructure.Repositories;
+
+[InstanceScopedService]
+public class GroupRepositoryAsync : IGroupRepositoryAsync
+{
+    private readonly IStudioManagementSystemDbContextAsync _smsDbContext;
+    
+    public GroupRepositoryAsync(IStudioManagementSystemDbContextAsync smsDbContext)
+    {
+        _smsDbContext = smsDbContext;
+    }
+    
+    public async Task<List<Group>> GetGroupsAsync(CancellationToken ct)
+    {
+        var groups = await _smsDbContext.Groups.ToListAsync(ct);
+        return groups;
+    }
+
+    public async Task<Group?> GetGroupAsync(Guid id, CancellationToken ct)
+    {
+        var group = await _smsDbContext.Groups.FindAsync(id, ct);
+        return group;
+    }
+
+    public async Task<Guid> AddGroupAsync(Group group, CancellationToken ct)
+    {
+        try
+        {
+            await _smsDbContext.Groups.AddAsync(group, ct);
+            await _smsDbContext.SaveChangesAsync(ct);
+        }
+        catch (Exception ex)
+        {
+            return Guid.Empty;
+        }
+
+        return group.Id;
+    }
+
+    public async Task<Group?> UpdateGroupAsync(Guid id, Group group, CancellationToken ct)
+    {
+        var myGroup = await _smsDbContext.Groups.FindAsync(id, ct);
+        try
+        {
+            myGroup.Name = group.Name;
+            await _smsDbContext.SaveChangesAsync(ct);
+        }
+        catch (Exception ex)
+        {
+            return null;
+        }
```

Comment:
the find async may throw (same with the method that repeats this logic). Let's re-use the existing method, check we got the group, and add some logging

```suggestion
        var group = await GetGroupAsync(id, ct)
            ?? throw new DataException($"No {nameof(Group)} was found with the given ID '{id}'");

        try
        {
            group.Name = name;
            await _smsDbContext.SaveChangesAsync(ct);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, $`an error occurred while updating the {nameof(Group)} name`)
            return false;
        }
```

Reply:
This means if we screw up in the rest of the app we'll quickly know,

* did we remember the ID?
* OR what was the unknown error we encountered?

Reply:
I'd repeat this logging pattern in particular around the repo's